### PR TITLE
Use secrets for providing MySQL password

### PR DESCRIPTION
**Pull request by aswietlik**

### CODE REVIEW

File: artifacts/mysql-statefulset.yaml [hunk 1]
```diff
@@ -128,24 +143,24 @@ spec:
...
             echo "Initializing replication from clone position"
             # In case of container restart, attempt this at-most-once.
             mv change_master_to.sql.in change_master_to.sql.orig
-            mysql -h 127.0.0.1 <<EOF
+            mysql -h 127.0.0.1 -p${MYSQL_ROOT_PASSWORD} <<EOF
           $(<change_master_to.sql.orig),
             MASTER_HOST='{{ .Name }}-0.{{ .Name }}',
             MASTER_USER='root',
-            MASTER_PASSWORD='',
+            MASTER_PASSWORD=${MYSQL_ROOT_PASSWORD},
```
Comment (earlgreyz) [inline artifacts/mysql-statefulset.yaml]: Missing `''` - this should be a SQL string 
```
MASTER_PASSWORD='${MYSQL_ROOT_PASSWORD}',
```

**Comment by aswietlik** [comment]
Fixed, everything should work now. Additionally, I added a better way of providing password in liveness/readiness probes.

File: artifacts/mysql-statefulset.yaml [hunk 2]
```diff
@@ -128,24 +137,24 @@ spec:
...
+            MASTER_PASSWORD='${MYSQL_ROOT_PASSWORD}',
             MASTER_CONNECT_RETRY=10;
           START SLAVE;
           EOF
           fi
 
           # Start a server to send backups when requested by peers.
           exec ncat --listen --keep-open --send-only --max-conns=1 3307 -c \
-            "xtrabackup --backup --slave-info --stream=xbstream --host=127.0.0.1 --user=root"
+            "xtrabackup --backup --slave-info --stream=xbstream --host=127.0.0.1 --user=root -p${MYSQL_ROOT_PASSWORD}"
```
Comment (earlgreyz) [inline artifacts/mysql-statefulset.yaml]: Maybe `--password=${MYSQL_ROOT_PASSWORD}`, to keep consistent flags usage.

File: artifacts/mysql-statefulset.yaml [hunk 3]
```diff
@@ -87,19 +90,25 @@ spec:
             memory: 1Gi
         livenessProbe:
           exec:
-            command: ["mysqladmin", "ping"]
+            command: ["bash", "-c", "mysqladmin ping -p$MYSQL_ROOT_PASSWORD"]
```
Comment (earlgreyz) [inline artifacts/mysql-statefulset.yaml]: Note that we might need to surround the variable with apostrophes `'${MYSQL_ROOT_PASSWORD}'` or `"${MYSQL_ROOT_PASSWORD}"` in order for it to work when password has whitespace characters.

Comment (aswietlik) [reply artifacts/mysql-statefulset.yaml]: Done. FYI: ```"${MYSQL_ROOT_PASSWORD}"``` version doesn't work.

File: artifacts/mysql-statefulset.yaml (hunk 2 above)
Comment (aswietlik) [reply artifacts/mysql-statefulset.yaml]: Done.

**Pull request merged**